drivers/aks/aks_driver.go

```diff
@@ -92,134 +158,201 @@ func (d *Driver) GetDriverCreateOptions(ctx context.Context) (*types.DriverFlags
 	driverFlag := types.DriverFlags{
 		Options: make(map[string]*types.Flag),
 	}
+
+	driverFlag.Options["subscription-id"] = &types.Flag{
+		Type:  types.StringType,
+		Usage: "Subscription credentials which uniquely identify Microsoft Azure subscription.",
+	}
+	driverFlag.Options["resource-group"] = &types.Flag{
+		Type:  types.StringType,
+		Usage: "The name of the 'Cluster' resource group.",
+	}
 	driverFlag.Options["name"] = &types.Flag{
 		Type:  types.StringType,
-		Usage: "The internal name of the cluster in Rancher",
+		Usage: "The name of the 'Cluster' resource, and the internal name of the cluster in Rancher.",
 	}
-	driverFlag.Options["display-name"] = &types.Flag{
+
+	driverFlag.Options["aad-client-app-id"] = &types.Flag{
 		Type:  types.StringType,
-		Usage: "The name that is displayed to the user on the Rancher UI",
+		Usage: `The ID of an Azure Active Directory client application of type "Native". This application is for user login via kubectl.`,
 	}
-	driverFlag.Options["subscription-id"] = &types.Flag{
+	driverFlag.Options["aad-server-app-id"] = &types.Flag{
 		Type:  types.StringType,
-		Usage: "Subscription credentials which uniquely identify Microsoft Azure subscription",
+		Usage: `The ID of an Azure Active Directory server application of type "Web app/API". This application represents the managed cluster's apiserver (Server application).`,
 	}
-	driverFlag.Options["resource-group"] = &types.Flag{
+	driverFlag.Options["aad-server-app-secret"] = &types.Flag{
 		Type:  types.StringType,
-		Usage: "The name of the resource group",
+		Usage: `The secret of an Azure Active Directory server application.`,
 	}
-	driverFlag.Options["location"] = &types.Flag{
+	driverFlag.Options["aad-tenant-id"] = &types.Flag{
 		Type:  types.StringType,
-		Usage: "Resource location",
+		Usage: `The ID of an Azure Active Directory tenant.`,
+	}
+
+	driverFlag.Options["enable-http-application-routing"] = &types.Flag{
+		Type:  types.BoolType,
+		Usage: `Enable the Kubernetes ingress with automatic public DNS name creation.`,
 		Default: &types.Default{
-			DefaultString: "eastus",
+			DefaultBool: true,
 		},
 	}
-	driverFlag.Options["tags"] = &types.Flag{
-		Type:  types.StringSliceType,
-		Usage: "Resource tags. For example, foo=bar",
+	driverFlag.Options["enable-monitoring"] = &types.Flag{
```

Comment:
I'm concerned about this option.   If this value was missing in previous versions what did Azure default the clusters to when we created them?

If it enabled monitoring I think this field should be a bool pointer type so that we can default nil values to true (when we parse the state from driver options).

Reply:
@nathan-jenan-rancher , apologize for my mistake, I have changed all `bool pointer type` to `bool type` without deep thought.

I prefer to use `bool pointer type` previously, because I think `enable-monitoring` and `enable-http-application-routing` are two new options, they haven't been visible to the previous created clusters. 

We need default nil values to nil, util passing `true` or `false` explicitly. For example, if a previous created cluster want to increase the cluster node count, `enable-xxx` are still nil to do nothing. If a new cluster want to enable moniotring, just pass `true` from UI or cli. 

Is it OK?

Reply:
Let's jump on zoom as soon as you get online and discuss.